tests/test_internals/test_si.py

```diff
@@ -576,5 +585,115 @@ def test_group_set_custom_description(self):
+
+        expected_channel_names_in_electrodes_table = ["0", "1", "2", "3", "4", "5"]
+        actual_channel_names_in_electrodes_table = list(self.nwbfile.electrodes["channel_name"].data)
+        self.assertListEqual(actual_channel_names_in_electrodes_table, expected_channel_names_in_electrodes_table)
+
+    def test_string_channel_names(self):
+        """Ensure channel names merge correctly after appending when channel names are strings."""
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile)
+
+        expected_channel_names_in_electrodes_table = ["a", "b", "c", "d", "e", "f"]
+        actual_channel_names_in_electrodes_table = list(self.nwbfile.electrodes["channel_name"].data)
+        self.assertListEqual(actual_channel_names_in_electrodes_table, expected_channel_names_in_electrodes_table)
+
+    def test_common_property_extension(self):
+        """Add a property for a first recording that is then extended by a second recording."""
+        self.recording_1.set_property(key="common_property", values=["value_1"] * self.num_channels)
+        self.recording_2.set_property(key="common_property", values=["value_2"] * self.num_channels)
+
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile)
+
+        actual_properties_in_electrodes_table = list(self.nwbfile.electrodes["common_property"].data)
+        expected_properties_in_electrodes_table = ["value_1", "value_1", "value_1", "value_1", "value_2", "value_2"]
+        self.assertListEqual(actual_properties_in_electrodes_table, expected_properties_in_electrodes_table)
+
+    def test_new_property_addition(self):
+        """Add a property only available in a second recording."""
+        self.recording_2.set_property(key="added_property", values=["added_value"] * self.num_channels)
+
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile)
+
+        actual_properties_in_electrodes_table = list(self.nwbfile.electrodes["added_property"].data)
+        expected_properties_in_electrodes_table = ["", "", "added_value", "added_value", "added_value", "added_value"]
+        self.assertListEqual(actual_properties_in_electrodes_table, expected_properties_in_electrodes_table)
+
+    def test_manually_added_before_recording(self):
+        """."""
+        # TODO, fill in any other required values to add_electrode
+        self.nwbfile.add_electrode(id=123)
+        self.nwbfile.add_electrode(id=124)
+
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile)
+
+        expected_ids = []
+        expected_names = []
+        self.assertListEqual(list(self.nwbfile.electrodes.id.data), expected_ids)
+        self.assertListEqual(list(self.nwbfile.electrodes["channel_name"].data), expected_names)
```

Comment:
@h-mayorquin Here is the base use-case logic we discussed.

Also, it's fine to have multiple assertions in a single function if they correspond to different expected outputs, its just best practice to keep the number of such assertions relatively small and/or vectorized as much as possible where appropriate.